packages/res/git-data.mjs

```diff
@@ -0,0 +1,24 @@
+import { execSync } from 'child_process';
+import * as fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+let x1 = 'unknown', x2 = 'unknown';
+
+// Retrieve GitHub branch and commit hash
+x1 = execSync('git rev-parse --abbrev-ref HEAD').toString().trim();
+x2 = execSync('git rev-parse --short HEAD').toString().trim();
+
+// Converting JS object to JSON string
+var jsonData = JSON.stringify({branch: x1, hash: x2});
+
+// Get current directory.
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// Write data in json file.
+fs.writeFile(__dirname + '/git-data.json', jsonData, (err) => {
```

Comment:
If you've already imported `path`, might as well use `path.join` here instead of + for obtaining the full path.